src/pretix/static/pretixcontrol/js/ui/main.js

```diff
@@ -742,3 +742,7 @@ $(function () {
     $("#ajaxerr").on("click", ".ajaxerr-close", ajaxErrDialog.hide);
     moment.locale($("body").attr("data-datetimelocale"));
 });
+
+$(document).ready(function () {
+    $('.form-horizontal').areYouSure( {'message': gettext('You have unsaved changes!')});
```

Comment:
Hmm, I'm not sure if using ``form-horizontal`` really follows the principle of least surprise and won't come back to bite us in the future. For example, this would not make it trigger for the "Comment" field on the order detail page, which is a field you really easily forget to save.

To exclude all search fields, we could use ``form[method=post]``. This might cause some problems with hidden/AJAX-enabled forms, so maybe we could add a ``.filter`` statement to get only those which have a submit button?

```
$('.form-horizontal').filter(function () { return $(this).find("button:not([type=button]), input[type=submit]").length > 0; })…
```

Haven't tested this yet.

Reply:
This actually seems to be working quite nicely! Did a few tests in the frontend and couldn't find any fields, where it misfired.